.github/actions/build_and_coverage/action.yml

```diff
@@ -6,14 +6,17 @@ runs:
   steps:
 
       # SETUP & CACHE
-    - uses: coursier/setup-action@v1.3.0
+    - uses: coursier/cache-action@v6.4.5
+      id: coursier-cache
       with:
-        jvm: adopt:8
-        apps: sbt scala scalafmt  # https://github.com/coursier/apps
+        ignoreJob: 'true'
+        ignoreMatrix: 'true'
 
-    - uses: coursier/cache-action@v6
+    - uses: coursier/setup-action@v1.3.5
+      if: steps.coursier-cache.outputs.cache-hit-coursier != 'true'
       with:
-        extraKey: "${{ env.SPARK_VERSION }}"
+        jvm: adopt:8
```

Comment:
something is not working fine here
before --> `[info] welcome to sbt 1.9.9 (AdoptOpenJDK Java 1.8.0_292)`
after --> `[info] welcome to sbt 1.9.9 (Eclipse Adoptium Java 11.0.22)`

Reply:
Is using a different JVM but is it affecting something?

Reply:
I thought cache won't affect Java, but it seems if cache works Java is not set (we are now skipping the "install" step)

Reply:
Could it be that you exchange the cache step with the prepare step? It's something that I see suspicious

Reply:
no, it is supposed to execute cache before the install so you avoid this step, that should not be the problem. The problem is that if we match a cache we shouldn't install anything else, I mean this section `apps: sbt scala scalafmt`. As we don't need to install those apps because they should be provided by the cache, I ignored the step completely, but the step not only install those apps, but it set the java version as well (`jvm: adopt:8`). Because the whole step is now ignored if there is a cache hit, the java set is ignored as well, so I would probably add a new step only setting the java version if there is a cache hit... but it is kind of weird and I'm trying to determine if cache is needed at all, because our build times are not much better with the cache step